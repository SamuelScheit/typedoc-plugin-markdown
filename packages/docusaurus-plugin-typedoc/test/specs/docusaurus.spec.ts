import * as fs from 'fs';
import * as path from 'path';
import typedocPlugin from '../../dist/plugin';

async function bootstrap(outDir: string, customOptions = {}) {
  const options = {
    id: 'default',
    out: outDir,
    entryPoints: ['./test/stubs/*.ts'],
    tsconfig: ['./test/stubs/tsconfig.json'],
    readme: 'none',
  };

  const plugin = typedocPlugin(
    {
      siteDir: path.join(__dirname, '..', 'out'),
      generatedFilesDir: '',
      siteConfig: {},
    },
    {
      ...options,
      ...customOptions,
    },
  );
  return await plugin;
}

describe(`Plugin:`, () => {
  describe(`Defaults`, () => {
    let tmpobj;
    beforeAll(async () => {
      await bootstrap('./test/out/default', { sidebar: { collapsed: false } });
    });
    test(`should render`, () => {
      const contents = fs
        .readFileSync(path.join(__dirname, '../out/default/index.md'))
        .toString();
      expect(contents).toMatchSnapshot();
    });

    test(`should generate typedoc sidebar`, async () => {
      const contents = fs
        .readFileSync(
          path.join(__dirname, '../out/default/typedoc-sidebar.cjs'),
        )
        .toString();
      expect(contents).toMatchSnapshot();
    });
  });
});
