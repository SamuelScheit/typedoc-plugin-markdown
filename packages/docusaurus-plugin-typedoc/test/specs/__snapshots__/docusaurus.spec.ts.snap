// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Docusaurus: Defaults should render docs 1`] = `
"# docusaurus-plugin-typedoc

## Modules

- [module-1](module-1/index.md)
- [module-2](module-2/index.md)
"
`;

exports[`Docusaurus: Defaults should render sidebar 1`] = `
"// @ts-check
/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
const typedocSidebar = { items: [{"type":"category","label":"module-1","items":[{"type":"category","label":"Classes","items":[{"type":"doc","id":"out/default/module-1/classes/ClassA","label":"ClassA"},{"type":"doc","id":"out/default/module-1/classes/ClassB","label":"ClassB"}]},{"type":"category","label":"Interfaces","items":[{"type":"doc","id":"out/default/module-1/interfaces/InterfaceA","label":"InterfaceA"},{"type":"doc","id":"out/default/module-1/interfaces/InterfaceB","label":"InterfaceB"}]}],"link":{"type":"doc","id":"out/default/module-1/index"}},{"type":"category","label":"module-2","items":[{"type":"category","label":"Classes","items":[{"type":"doc","id":"out/default/module-2/classes/ClassA","label":"ClassA"},{"type":"doc","id":"out/default/module-2/classes/ClassB","label":"ClassB"}]},{"type":"category","label":"Interfaces","items":[{"type":"doc","id":"out/default/module-2/interfaces/InterfaceA","label":"InterfaceA"},{"type":"doc","id":"out/default/module-2/interfaces/InterfaceB","label":"InterfaceB"}]}],"link":{"type":"doc","id":"out/default/module-2/index"}}]};
module.exports = typedocSidebar.items;"
`;

exports[`Docusaurus: Global Members should render docs 1`] = `
"# docusaurus-plugin-typedoc

## Classes

- [ClassA](classes/ClassA.mdx)
- [ClassB](classes/ClassB.mdx)

## Interfaces

- [InterfaceA](interfaces/InterfaceA.mdx)
- [InterfaceB](interfaces/InterfaceB.mdx)
"
`;

exports[`Docusaurus: Global Members should render sidebar 1`] = `
"// @ts-check
/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
const typedocSidebar = { items: [
  {
    "type": "category",
    "label": "Classes",
    "items": [
      {
        "type": "doc",
        "id": "out/global-members/classes/ClassA",
        "label": "ClassA"
      },
      {
        "type": "doc",
        "id": "out/global-members/classes/ClassB",
        "label": "ClassB"
      }
    ]
  },
  {
    "type": "category",
    "label": "Interfaces",
    "items": [
      {
        "type": "doc",
        "id": "out/global-members/interfaces/InterfaceA",
        "label": "InterfaceA"
      },
      {
        "type": "doc",
        "id": "out/global-members/interfaces/InterfaceB",
        "label": "InterfaceB"
      }
    ]
  }
]};
module.exports = typedocSidebar.items;"
`;
